### ISSUE TITLE
Joe's Review

### ISSUE THREAD

**Issue by jgrassel** [OpenLiberty/draft-guide-jpa-intro#32]
My thoughts, comments, and suggestions:

**What You'll Learn**

1) What level of expertise are we expecting of the user?  I don't see CRUD defined anywhere near the top, so are we expecting a base level of database knowledge?

2) Might sound better? " eliminates the burden of having to write **JDBC and** SQL code when performing database operations."

3) "The application that you will be working with is the event manager, which provides services for creating, retrieving, updating, and deleting event **entities**." -- In this statement, are you referring to JPA entity types?  I ask because the word "entity" gets to be a really overloaded term, so if another word, to diminish confusion, can be used that might be better

4) "You will use JPA annotations to define an entity class whose fields are persisted to the database. Every entity object is managed by an entity manager that will be injected into the Enterprise Java Bean (EJB)."

Maybe something more like: "You will use JPA annotations to define an entity class whose fields are persisted to the database.  The interaction between your application and the database is mediated the persistence context managed by an EntityManager, which itself is injected into the Enterprise Java Bean (EJB) Session Bean."

5) "In this guide, you will learn how to use a container-managed entity manager which means the EJB container is responsible for opening and closing the entity manager, as well as managing the transaction boundaries so that you do not need to explicitly begin, commit, or rollback database transactions."

Maybe something more like: "In this guide, you will learn how to use a _(Transaction-Scoped)_ Container-Managed Persistence Context EntityManager (which places the burden of managing the EntityManager's lifecycle on the application server) with a _(Stateless)_ EJB Session Bean; the Session Bean has been configured so that the EJB Container manages the global transaction boundaries, so that you do not need to explicitly begin, commit, or rollback database transactions."

**Getting Started**

**Defining a JPA entity class**
1) "JPA uses a database table for every entity" -- technically it's more complicated than that, when you consider the various types of inheritance schemes, secondary table mappings, etc.

2) Looking at the Event entity class, I gotta ask: A String for time Data?
```
@Column(name = "eventTime")
    private String time;
```
   Why not use java.util.Date (with @ Temporal) or java.sql.Timestamp?

**Configuring the entity manager**

1) I see lots of use of "entity manager" throughout the document, let's actually use its proper class name, "EntityManager".

2) "An entity manager provides the functionality for performing operations on the database such as persisting, loading, updating, and deleting entities."

Maybe change that to "An EntityManager provides the functionality to perform operations on the database such as persisting new entities (table rows), fetching entities through find() and queries, updating changes, and removing entities (table rows.)"

3) In an EE environment, you do not need the `<class>io.openliberty.guides.jpaguide.models.Event</class>` entry.  That is for only for the JSE environment (or applications which use the JSE bootstrapping method, something I try to discourage.)

**Performing CRUD operations using JPA**

1) Ok, so you define CRUD here.  I think it should be defined much earlier, so the reader isn't distracted what CRUD stands for, since the term is introduced right at the beginning.

2) Though the default is Container Managed Transactions, it might be helpful to annotate the session bean with the @TransactionManagement(CONTAINER)  annotation, and session bean's business methods with the @TransactionAttribute(REQUIRED) annotation (the defaults).  We shouldn't assume the reader knows those are the defaults.

3) "Since the entity manager is container-managed, the EJB container will ensure that the methods are executed within an active transaction which is why it is not necessary to explicitly begin, commit, or rollback transactions. "

No.  "Since the Stateless Session Bean is configured to manage the global transaction, the EJB Container will ensure that its business methods are executed within the boundaries of an active global transaction, which is why it is not necessary to explicitly begin, commit, or rollback transactions."

A Container Managed Persistence Context's EntityManager will enlist with the current global transaction (caveat: since 2.1 that can be made optional given the new transaction synchronization configurability.)  The EntityManager itself doesn't start or commit/rollback transactions (though it can mark the global transaction for rollback given certain error events.)

4) "At the end of an EJB operation, the persistence context will be notified that the transaction is committing and the persistence provider, EclipseLink, will flush the changes to the database."

I think this might sound better "At the end of a session bean business method invocation, the persistence context will be notified that the transaction is committing, and will flush any changes made to entity instances it is managing to the database."

It is important to clarify this, because if you em.detach() entities before the commit, changes made to those entities would be lost.  You would have to invoke a manual flush with em.flush() before detaching those entities if they have changes you want to save on the database.

5) Somewhere here might be a good idea to point out that at the end of transaction boundaries, entities managed by a Container Managed Persistence Context become detached automatically.

It might be a good idea to dedicate a paragraph explaining the notion of new entities (constructed, never persisted by an EntityManager), managed entities (new entities that have been persist()ed, fetched by find() and queries, and merge()d, that are being actively tracked by the persistence context), and detached entities (entities that have been explicitly kicked out of the persistence context by em.clear() or em.detach(), or in a CMTS persistence context, a commit/rollback transaction boundary.)

Of course, things get more complicated when you consider persistence context propagation, which gets even more hairier when you consider transaction suspension when encountering business methods annotated with REQUIRES_NEW, etc.

6) "The updateEvent() method updates an instance of the Event entity class in the database. The EntityManager automatically tracks entity objects in the persistence context for changes and synchronizes them with the database. The merge() method will copy the entity supplied as an argument into a new Event instance that will be added to the persistence context so that its changes can be tracked."

Yeah, that wording bugs me.  It comes back to the necessity of explaining new/managed/detached entities.  The thing is, changes to managed entities are automatically tracked by the persistence context, you don't have to do anything.  But if your entity becomes detached, then yes, you have to merge that entity into the current persistence context (it gets hairier when you consider that the merge will only track changes for fields which have been loaded before the entity was detached, so there's ramifications for LAZY loaded fields here.)

7) "The deleteEvent() method removes an instance of the Event entity class from the database by calling the remove() method on an EntityManager instance. An instance of an entity class must be passed as an argument to the remove() method, otherwise an IllegalArgumentException will be thrown."

Eh, maybe say it marks the entity for deletion, because it doesn't actually issue the DELETE FROM SQL statement until either a flush() (implicit or explicit, implicit flushes can happen any time you execute a query) or at transaction commit.  Considering that remove() requires an active transaction in this scenario, that detail might not matter

(NOTE: I haven't executed the test yet, will update this issue when I have)
